src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -974,6 +973,26 @@ export abstract class GridTableBase<T> extends Disposable implements IView, IQue
 		];
 	}
 
+	private isJsonCell(value: ICellValue): boolean {
+		return !!(value && !value.isNull && value.displayValue?.match(IsJsonRegex));
+	}
+
+	private isXmlCell(value: ICellValue): boolean {
+		let isXML = false;
+		try {
+			if (value && !value.isNull && value.displayValue.trim() !== '') {
+				var parser = new DOMParser();
```

Comment:
Something to think about - what's the performance like on this? Especially for large data sets with large string values. We may want to consider adding some sort of limitation (such as max string length) and/or a setting to enable/disable the parsing to avoid problems caused by bad performance

Reply:
This is also in my thoughts. The next thing I'm thinking of implementing is a way for customers to configure whether they want datatype detection on column-basis or cell-basis. As currently, when rendering slick-grid and formatting data for XML/JSON, we determine datatype of each cell but update it for entire column, that leads to lots of problems if same column is having dynamic datatypes.

If users want to go for cell-wise type detection, it will be slow to perform, but we will make sure we don't handle the datatype column-wide and data will be formatted on cell-basis for all rows. But if customers don't have this scenario (which should be ideal / default), we will only detect datatypes in first non-empty column for XML/JSON and as soon as it's determined, not worry about all remaining data - this should help us gain lot of perf benefit when rendering.